app/views/user_sessions/_form.html.erb

```diff
@@ -1,110 +1,104 @@
 
-  <input
-    type="hidden"
-    name="return_to"
-    class="login-modal-redirect"
-    value="<%= request.fullpath %>"
-  />
-
-  <center>
-    <h2><%= raw t('user_sessions.new.log_in') %></h2>
-    <p class="form-grey" id="toSignup">or
+<input type="hidden" name="return_to" class="login-modal-redirect" value="<%= request.fullpath %>" />
+
+<center>
+  <h2><%= raw t('user_sessions.new.log_in') %></h2>
+  <p class="form-grey" id="toSignup">or
     <%= link_to signup_path( return_to: params[:return_to] ) do %>
-      <%= raw t('user_sessions.new.sign_up') %></a> <%= raw t('user_sessions.new.to_join') %>
+    <%= raw t('user_sessions.new.sign_up') %></a> <%= raw t('user_sessions.new.to_join') %>
     <% end %></p>
-  </center>
+</center>
 
-  <% if f.error_messages != "" %><div class="alert alert-danger"><%= f.error_messages %></div><% end %>
+<% if f.error_messages != "" %><div class="alert alert-danger"><%= f.error_messages %></div><% end %>
 
-  <br style="clear:both;"/>
+<br style="clear:both;" />
 
-  <div id="socialIcons" class='col-lg-12'>
-    <%= render :partial => "layouts/social_icons" %>
-  </div>
+<div id="socialIcons" class='col-lg-12'>
+  <%= render :partial => "layouts/social_icons" %>
+</div>
 
-  <br />
+<br />
 
-  <br style="clear:both;"/>
+<br style="clear:both;" />
 
-  <div id="toggle" class="login-form">
+<div id="toggle" class="login-form">
 
-    <div class="form-group">
-      <label for="username"><%= t('user_sessions.new.username') %></label>
-      <%= f.text_field :username, { tabindex: 1, placeholder: "Username", class: 'form-control', id: 'username-login', required: true } %>
+  <div class="form-group">
+    <label for="username"><%= t('user_sessions.new.username') %></label>
+    <%= f.text_field :username, { tabindex: 1, placeholder: "Username", class: 'form-control', id: 'username-login', required: true } %>
 
-      <div class="form-group has-feedback">
-        <label class="control-label" for="password"><%= t('user_sessions.new.password') %></label>
-        <div class="input-group">
-          <%= f.password_field :password, { tabindex: 2,placeholder: "Password",class: 'form-control', id: 'password-signup', onpaste: 'return false;' , required: true } %>
-          <div class="input-group-append">
-            <a
-              id="visibility"
-              class="fa fa-eye input-group-text"
-              aria-hidden="true">
-            </a>
-          </div>
+    <div class="form-group has-feedback">
+      <label class="control-label" for="password"><%= t('user_sessions.new.password') %></label>
+      <div class="input-group">
+        <%= f.password_field :password, { tabindex: 2,placeholder: "Password",class: 'form-control', id: 'password-signup', onpaste: 'return false;' , required: true } %>
+        <div class="input-group-append">
+          <a id="visibility" class="fa fa-eye input-group-text d-flex align-items-center" aria-hidden="true">
```

Comment:
The real change here is that `d-flex` class has been added to the `a#visibility` element.
I can confirm that this works and solves misalignment problem on all pages.

Reply:
Is this solved?

Reply:
Yes, it is.